Documentation/gettingstarted/cni-chaining-aws-cni.rst

```diff
@@ -6,69 +6,88 @@
-plugin is installed. If you have set up an EKS cluster, this is automatically
-done.
+Ensure that the `aws-vpc-cni-k8s <https://github.com/aws/amazon-vpc-cni-k8s>`_
+plugin is installed — which will already be the case if you have created an EKS
+cluster. Also, ensure the version of the plugin is up-to-date as per the above.
 
 .. include:: k8s-install-download-release.rst
 
-Deploy Cilium release via Helm:
+Deploy Cilium via Helm:
 
 .. parsed-literal::
 
    helm install cilium |CHART_RELEASE| \\
      --namespace kube-system \\
      --set cni.chainingMode=aws-cni \\
-     --set enableIPv4Masquerade=false \\
+     --set masquerade=false \\
      --set tunnel=disabled \\
      --set nodeinit.enabled=true
 
-This will enable chaining with the aws-cni plugin. It will also disable
-tunneling. Tunneling is not required as ENI IP addresses can be directly routed
-in your VPC. You can also disable masquerading for the same reason.
+This will enable chaining with the AWS VPC CNI plugin. It will also disable
+tunneling, as it's not required since ENI IP addresses can be directly routed
+in the VPC. For the same reason, masquerading can be disabled as well.
 
 Restart existing pods
 =====================
 
-The new CNI chaining configuration will *not* apply to any pod that is already
-running in the cluster. Existing pods will be reachable and Cilium will
-load-balance to them but policy enforcement will not apply to them and
-load-balancing is not performed for traffic originating from existing pods.
-You must restart these pods in order to invoke the chaining configuration on
-them.
+The new CNI chaining configuration *will not* apply to any pod that is already
+running in the cluster. Existing pods will be reachable, and Cilium will
+load-balance *to* them, but not *from* them. Policy enforcement will also not
+be applied. For these reasons, you must restart these pods so that the chaining
+configuration can be applied to them.
+
+The following command can be used to check which pods need to be restarted:
+
+.. code:: bash
 
-If you are unsure if a pod is managed by Cilium or not, run ``kubectl get cep``
-in the respective namespace and see if the pod is listed.
+   for ns in $(kubectl get ns -o jsonpath='{.items[*].metadata.name}'); do
+        ceps=$(kubectl -n "${ns}" get cep \
+            -o jsonpath='{.items[*].metadata.name}')
+        pods=$(kubectl -n "${ns}" get pod \
+            -o custom-columns=NAME:.metadata.name,NETWORK:.spec.hostNetwork \
+            | grep -E '\s(<none>|false)' | awk '{print $1}' | tr '\n' ' ')
+        ncep=$(echo "${pods} ${ceps}" | tr ' ' '\n' | sort | uniq -u | paste -s -d ' ' -)
+        for pod in $(echo $ncep); do
+          echo "${ns}/${pod}";
+        done
+   done
 
 .. include:: k8s-install-validate.rst
 
+.. important::
+
+   As mentioned above, some advanced Cilium features may be limited when
+   chaining with other CNI plugins. Hence, at least one of the connectivity
+   tests above may fail.
```

Comment:
This reads like we expect `at least` one check to fail. In that case, we should probably name it. Otherwise, I would phrase it differently, `Hence, some of the connectivity tests above may fail`.

Reply:
Makes total sense :+1: